lib/kvdb/kvdb_rest.c

```diff
@@ -1093,8 +1101,10 @@ kvs_query_tree(
         }
 
         if (view->eklen > 0) {
-            for (size_t i = 0; i < sizeof(view->ekbuf); i++)
-                snprintf(kbuf + i, sizeof(kbuf) - i, "%x", view->ekbuf[i]);
+            char kbuf[sizeof(view->ekbuf) * 3 + 1];
+
+            fmt_hexp(kbuf, sizeof(kbuf), view->ekbuf,
+                     min_t(size_t, view->eklen, sizeof(view->ekbuf)), "", 2, ".", "");
```

Comment:
are you okay with using `fmt_hex`?  If so, use `fmt_hex_buf_size` to size kbuf (or is that even possible for local var init?).

if you want `fmt_hexp`, consider adding `fmt_hexp_buf_size` to do the math for you.

Reply:
The markers between each two bytes of the key make it much more readable IMO.  With the separators I can more easily convert ascii codes to letters in my head and keep track of where I'm looking.  Without them I get totally lost.

Oh, I see, fmt_hex() uses dashes. I don't mind the dashes, but the 0x at the beginning seems unnecessary...

```
7573.6572.3838.3437.3133.3239.3439.3036.3138.3637
7573-6572-3838-3437-133-3239-3439-3036-3138-3637
7573657238383437313332393439303631383637
```

Reply:
I'm fine w/ changing `fmt_hex`